api/v1_projects.inc

```diff
@@ -271,6 +313,47 @@ function render_project_page_json($row)
     ];
 }
 
+function render_page_data($lpage)
+{
+    global $projects_url;
+    return [
+        "text" => $lpage->get_text(),
+        "image_url" => "$projects_url/{$lpage->projectid}/{$lpage->imagefile}",
+        "page_state" => $lpage->page_state,
+        "canRestore" => $lpage->can_be_reverted_to_last_save(),
```

Comment:
I agree it would be better for the caller to handle it. It's only purpose is to disable the "restore to last save" button if the text hasn't yet been saved in the round; i.e. if the page is in page_out state. Is the caller allowed to interpret the meaning of the page state?
Alternatively we could not bother to disable the button. But instead copy the text from the "previous round" into the current round. Then if the user pressed the button before he had saved the text he would just get the original text rather than nothing.

Reply:
The problem is that if you reopen a page you worked on earlier it can be in temp state, meaning you saved it earlier or it could still be in out state if you never saved it but just stopped proofreading.